apiserver/facades/agent/upgrader/service.go

```diff
@@ -64,6 +66,34 @@ type CredentialService interface {
 	CloudCredential(ctx context.Context, key credential.Key) (cloud.Credential, error)
 }
 
+type ControllerNodeService interface {
+	SetReportedControllerNodeAgentVersion(context.Context, string, coreagentbinary.Version) error
```

Comment:
Comment me. Can we make string a named arg.

Strictly the interface names should have comments but so long as the methods do...

Reply:
Done & Done